modules/network/src/test/java/org/apache/ignite/network/internal/netty/InboundDecoderTest.java

```diff
@@ -87,6 +92,35 @@ public void test(long seed) throws Exception {
         assertEquals(msg, output);
     }
 
+    /**
+     * Tests that an {@link InboundDecoder} doesn't hang if it encounters a byte buffer with only partially written
+     * header.
+     *
+     * @throws InterruptedException If failed.
+     */
+    @Test
+    public void testPartialHeader() throws InterruptedException {
+        var registry = new MessageSerializationRegistry();
+
+        var channel = new EmbeddedChannel(new InboundDecoder(registry));
+
+        ByteBuf buffer = allocator.buffer();
+
+        buffer.writeByte(1);
+
+        var latch = new CountDownLatch(1);
+
+        new Thread(() -> {
+            channel.writeInbound(buffer);
```

Comment:
why do you need to run this in a separate thread?

Reply:
To check if it hangs

Reply:
ok, then it can be written a little bit shorter:
```
CompletableFuture
    .runAsync(() -> {
        channel.writeInbound(buffer);

        channel.readInbound();
    })
    .get(3, TimeUnit.SECONDS);
```